docs/persistence.md

```diff
@@ -0,0 +1,128 @@
+# Overview
+Cadence has a well defined API interface at the persistence layer. Any database that supports multi-row transactions on
+a single shard or partition can be made to work with cadence. This includes cassandra, dynamoDB, auroraDB, MySQL,
+Postgres and may others. There are currently two supported database implementations at the persistence layer - 
+cassandra and MySQL. This doc shows how to run cadence with cassandra and mysql. It also describes the steps involved
+in adding support for a new database at the persistence layer.
+ 
+# Getting started on mac
+## Cassandra
+### Start cassandra
+```
+brew install cassandra
+brew services start cassandra
+```
+### Install cadence schema
+```
+cd $GOPATH/github.com/uber/cadence
+make install-schema
+```
+
+### Start cadence server
+```
+cd $GOPATH/github.com/uber/cadence
+./cadence-server start --services=frontend,matching,history,worker
+```  
+ 
+## MySQL
+### Start MySQL server
+```
+brew install mysql
+brew services start mysql
+```
+### Install cadence schema
+```
+cd $GOPATH/github.com/uber/cadence
+make install-schema-mysql
+```
+
+### Start cadence server
+```
+cd $GOPATH/github.com/uber/cadence
+cp config/development_mysql.yaml config/development.yaml
+./cadence-server start --services=frontend,matching,history,worker
+```
+
+# Configuration
+## Common to all persistence implementations
+There are two major sub-subsystems within cadence that need persistence - cadence-core and visibility. cadence-core is
+the workflow engine that uses persistence to store state tied to domains, workflows, workflow histories, task lists 
+etc. cadence-core powers almost all of the cadence APIs. cadence-core could be further broken down into multiple 
+subs-systems that have slightly different persistence workload characteristics. But for the purpose of simplicity, we 
+don't expose these sub-systems today but this may change in future. Visibility is the sub-system that powers workflow 
+search. This includes APIs such as ListOpenWorkflows, ListClosedWorkflows or ListWorkflowsWithTag etc. Today, it is 
```

Comment:
Probably remove ListWorkflowsWithTag because we don't have it on Cassandra or MySQL.

Reply:
👍